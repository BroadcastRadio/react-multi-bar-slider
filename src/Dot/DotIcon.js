import PropTypes from 'prop-types';
import glamorous from 'glamorous';

const DotIcon = glamorous.img({
  position: 'absolute',
  transform: 'translateX(-50%)',
  userDrag: 'none',
  userSelect: 'none'
}, ({ width, height }) => ({
  width,
  height
}));

DotIcon.displayName = 'DotIcon';

DotIcon.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  css: PropTypes.object
};

DotIcon.defaultProps = {
  width: 50,
  height: 50
};

export default DotIcon;
