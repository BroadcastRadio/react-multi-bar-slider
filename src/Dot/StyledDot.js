import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import transition from '../utils/transition';

const StyledDot = styled('span')({
  position: 'absolute',
  right: 0,
  display: 'block',
  zIndex: 5,
  borderRadius: '50%'
}, ({ hasIcon, width, height, color, reversed, noTransition, css }) => ({
  top: hasIcon ? 0 : '50%',
  left: reversed ? 0 : 'auto',
  right: reversed ? 'auto' : 0,
  transform: hasIcon
    ? 'translateX(-50%)'
    : reversed ? 'translate(-50%, -50%)' : 'translate(50%, -50%)',
  width: hasIcon ? 0 : width,
  height: hasIcon ? 0 : height,
  backgroundColor: hasIcon ? 'transparent' : color,
  transition: noTransition ? 'none' : transition,
  ...css
}));

StyledDot.propTypes = {
  hasIcon: PropTypes.bool.isRequired,
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  color: PropTypes.string.isRequired,
  reversed: PropTypes.bool.isRequired,
  noTransition: PropTypes.bool.isRequired,
  css: PropTypes.object
};

StyledDot.defaultProps = {
  width: 28,
  height: 28,
  css: null
};

export default StyledDot;
