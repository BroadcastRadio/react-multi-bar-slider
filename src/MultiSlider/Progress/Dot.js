import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import transition from '../utils/transition';

const StyledDot = glamorous.span({
  position: 'absolute',
  display: 'block',
  zIndex: 5,
  borderRadius: '50%'
}, ({ hasIcon, width, height, color, noTransition }) => ({
  top: hasIcon ? 0 : '50%',
  transform: hasIcon ? 'translateX(-50%)' : 'translate(50%, -50%)',
  width: hasIcon ? 0 : width,
  height: hasIcon ? 0 : height,
  backgroundColor: hasIcon ? 'transparent' : color,
  transition: noTransition ? 'none' : transition
}));

const DotIcon = glamorous.img({
  position: 'absolute',
  transform: 'translateX(-50%)',
  userDrag: 'none',
  userSelect: 'none'
}, ({ width, height }) => ({
  width,
  height
}));

const Dot = ({ dot, sliderColor, mouseDown }) => (
  <StyledDot
    className="dot"
    hasIcon={!!dot.icon}
    width={dot.width || 28}
    height={dot.height || 28}
    color={dot.color || sliderColor}
    css={dot.style}
    noTransition={mouseDown}
  >
    {dot.icon && (
      <DotIcon
        className="icon"
        src={dot.icon}
        width={dot.width || 50}
        height={dot.height || 50}
        draggable={false}
        css={dot.iconStyle}
      />
    )}
  </StyledDot>
);

Dot.propTypes = {
  dot: PropTypes.oneOfType([
    PropTypes.shape({
      width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      color: PropTypes.string,
      icon: PropTypes.string,
      style: PropTypes.object
    }),
    PropTypes.bool
  ]).isRequired,
  sliderColor: PropTypes.string.isRequired,
  mouseDown: PropTypes.bool.isRequired
};

export default Dot;
