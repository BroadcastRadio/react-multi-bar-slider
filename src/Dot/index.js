import React from 'react';
import PropTypes from 'prop-types';
import processStyle from '../utils/processStyle';
import StyledDot from './StyledDot';
import DotIcon from './DotIcon';

const Dot = ({
  width,
  height,
  color,
  icon,
  style,
  iconStyle,
  sliderColor,
  reversed,
  mouseDown
}) => (
  <StyledDot
    className="dot"
    hasIcon={!!icon}
    width={width}
    height={height}
    color={color || sliderColor}
    reversed={reversed}
    noTransition={mouseDown}
    css={processStyle(style, {
      width,
      height,
      color,
      icon,
      sliderColor,
      reversed,
      mouseDown
    })}
  >
    {icon && (
      <DotIcon
        className="icon"
        src={icon}
        width={width}
        height={height}
        draggable={false}
        css={processStyle(iconStyle, {
          width,
          height,
          color,
          icon,
          sliderColor,
          mouseDown
        })}
      />
    )}
  </StyledDot>
);

Dot.displayName = 'Dot';

Dot.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  color: PropTypes.string,
  icon: PropTypes.string,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
  iconStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
  sliderColor: PropTypes.string,
  reversed: PropTypes.bool,
  mouseDown: PropTypes.bool
};

export default Dot;
