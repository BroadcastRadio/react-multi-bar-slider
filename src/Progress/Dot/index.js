import React from 'react';
import PropTypes from 'prop-types';
import processStyle from '../../utils/processStyle';
import StyledDot from './StyledDot';
import DotIcon from './DotIcon';

const Dot = ({ dot, sliderColor, reversed, mouseDown }) => (
  <StyledDot
    className="dot"
    hasIcon={!!dot.icon}
    width={dot.width}
    height={dot.height}
    color={dot.color || sliderColor}
    reversed={reversed}
    noTransition={mouseDown}
    css={processStyle(dot.style, { dot, sliderColor, reversed, mouseDown })}
  >
    {dot.icon && (
      <DotIcon
        className="icon"
        src={dot.icon}
        width={dot.width }
        height={dot.height}
        draggable={false}
        css={processStyle(dot.iconStyle, { dot, sliderColor, mouseDown })}
      />
    )}
  </StyledDot>
);

Dot.propTypes = {
  dot: PropTypes.oneOfType([
    PropTypes.shape({
      width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      color: PropTypes.string,
      icon: PropTypes.string,
      style: PropTypes.object
    }),
    PropTypes.bool
  ]).isRequired,
  sliderColor: PropTypes.string.isRequired,
  reversed: PropTypes.bool.isRequired,
  mouseDown: PropTypes.bool.isRequired
};

export default Dot;
