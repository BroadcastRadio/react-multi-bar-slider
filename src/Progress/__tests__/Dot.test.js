import React from 'react';
import Enzyme, { shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Dot from '../Dot';
import StyledDot from '../Dot/StyledDot';
import DotIcon from '../Dot/DotIcon';

Enzyme.configure({ adapter: new Adapter() });

beforeEach(() => {
  global.dotProps = {
    dot: {
      width: 40,
      height: 40,
      color: 'green',
      style: { color: 'purple' }
    },
    sliderColor: 'blue',
    reversed: false,
    mouseDown: false
  };
});

afterEach(() => {
  delete global.dotProps;
});

/* global dotProps */
describe('Dot.js', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(<Dot {...dotProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const dot = render(<Dot {...dotProps} />);
    expect(dot).toBeDefined();
  });

  it('renders a StyledDot', () => {
    const dot = shallow(<Dot {...dotProps} />);
    expect(dot.find(StyledDot).length).toBe(1);
  });

  it('renders a DotIcon when an icon is given', () => {
    const dot = shallow(<Dot {...dotProps} />);
    expect(dot.find(DotIcon).length).toBe(0);

    dotProps.dot.icon = 'foo.png';
    const dot2 = shallow(<Dot {...dotProps} />);
    expect(dot2.find(DotIcon).length).toBe(1)
    expect(dot2.find(DotIcon).prop('src')).toBe('foo.png');
  });
});
