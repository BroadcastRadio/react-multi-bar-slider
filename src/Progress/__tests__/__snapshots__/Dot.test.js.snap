// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot.js matches the snapshot 1`] = `
<span
  className="css-1mk3smr dot"
/>
`;
